decisions/backlog/0023-error-responses.md

```diff
@@ -0,0 +1,83 @@
+# 23. Action Authenticate: Error Responses (Proposal #1)
+
+Date: 2023-09-20
+
+## Status
+
+In Progress
+
+## Context
+
+The Tech. Specs. builds on existing standards for the overall Authentication flow. For the Action `Authenticate` specifically, the Tech. Specs. refer to the standard 'The OAuth 2.0 Authorization Framework' (https://www.rfc-editor.org/rfc/rfc6749). However, the requirement to implement the Action `Authenticate` based on this standard conflicts with the requirements in the Error Responses (V1: 6.5, V2: 6.9). As a result, it is theoretically impossible to implement a conformant solution. 
+
+#### Action Authenticate
+* **Tech. Specs. (both V1 and V2), Section 6.4.1/6.5:** *"Host systems MUST implement this action in conformance with [rfc6749] Section 4.4 (OAuth2 Client Credentials)."*
+* **RFC6749 (https://www.rfc-editor.org/rfc/rfc6749) Section 4.4:** *If the request failed client authentication or is invalid, the authorization server returns an error response as described in Section 5.2.*
+  * **Section 5.2:** *"The authorization server responds with an HTTP 400 (Bad Request) status code (unless specified otherwise) and includes the following parameters with the response:"*
+    *  "error (REQUIRED)"
+    *  "error_description (OPTIONAL)"
+    *  "error_uri (OPTIONAL)"
+
+For example:
+```javascript
+     HTTP/1.1 400 Bad Request
+     Content-Type: application/json;charset=UTF-8
+     Cache-Control: no-store
+     Pragma: no-cache
+
+     {
+       "error":"..."
+     }   
+```
+
+#### Error Responses
+* **Tech. Specs. (both V1 and V2), Section 6.5/6.9:**
+  * *"Whenever a host system returns an error response, it MUST send a HTTP response such that:"*
+    * *"with response body the error response"*
+  * *"A error response is a JSON object with the following properties:*
+    * *"code: a error response code encoded as a String"*
+    * *"message: a error message encoded as a String"*
+
+For example:
+```javascript
+     HTTP/1.1 400 Bad Request
+     Content-Type: application/json;charset=UTF-8
+     Cache-Control: no-store
+     Pragma: no-cache
+
+     {
+       "code":"...",
+       "message":"..."
+     }   
+```
+
+As shown, these requirements conflict when it comes to the structure of the body of the error response. Therefore, the Tech. Specs. need an update to eliminate this conflict.
```

Comment:
The tech specs refer to the RFC already quite explicitly, for my reading:

<img width="849" alt="Screenshot 2023-10-11 at 15 39 10" src="https://github.com/wbcsd/data-exchange-protocol/assets/38831/931b448d-a019-44d3-92e4-314d228bbfc6">

Would you agree?